File: Assignment/Abstraction/BinaryExpressionNode.cs
namespace Assignment.Abstraction
{
    internal struct BinaryExpressionNode : IExpressionNode
    {
        public BinaryExpressionNode(INode left, INode right, Operations operation)
        {
            this.Left = left;
            this.Right = right;
            this.Operation = operation;
        }

        public Operations Operation { get; }
        public INode Left { get; }
        public INode Right { get; }

        public string DisplayName()
        {
            return $"{nameof(BinaryExpressionNode)}(operation: {this.Operation})";
        }
    }
}

File: src/main/java/compiler/abstraction/expressions/BinaryExpressionNode.java
package compiler.abstraction.expressions;

import java.util.UUID;

import compiler.abstraction.ArithmeticOperation;
import compiler.abstraction.INode;

public class BinaryExpressionNode extends INode {
    public final INode left;
    public final INode right;
    public final ArithmeticOperation operation;

    public BinaryExpressionNode(INode left, INode right, ArithmeticOperation operation) {
        super();
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    public BinaryExpressionNode(INode left, INode right, ArithmeticOperation operation, UUID id) {
        super(id);
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(operation: %s)", BinaryExpressionNode.class.getSimpleName(), this.operation);
    }
}